source/Calamari.Common/Plumbing/Variables/VariablesFactory.cs

```diff
@@ -57,17 +57,17 @@ void LoadExecutionVariablesFromFiles(CommonOptions options)
                 
                 executionVariables = new CalamariExecutionVariableCollection();
 
-                LoadExecutionVariablesFromFile(options);
+                executionVariables.AddRange(LoadExecutionVariablesFromFile(options));
 
                 // This exists as the V2 pipeline stores both the parameters and the contents of the variables files for resiliency
                 // This should be removed once the first version this is deployed to has rolled out to most cloud customers
-                ReadDeprecatedVariablesFormatFromFiles(options);
+                executionVariables.AddRange(ReadDeprecatedVariablesFormatFromFiles(options));
 
-                ReadOutputVariablesFromOfflineDropPreviousSteps(options);
+                executionVariables.AddRange(ReadOutputVariablesFromOfflineDropPreviousSteps(options));
             }
         }
 
-        void LoadExecutionVariablesFromFile(CommonOptions options)
+        CalamariExecutionVariableCollection LoadExecutionVariablesFromFile(CommonOptions options)
```

Comment:
Changed these methods to be `Pure` rather than modifying a field.
Meant that it was easier for the nullable analyzer to understand what was going on